### DISCUSSION TITLE
Load simple static website project directory inside an IFRAME?

### DISCUSSION THREAD

**Discussion by browser-bug** [webpack/webpack#16016] [Q&A]
Hello everyone! I hope to be in the right place to ask this. I have the following structure inside my ReactTS project:
```
.src
├── App.tsx
├── external
│   ├── dashboard-app
│   │   ├── images
│   │   │   ├── avatars
│   │   │   │   └── user01.svg
│   │   │   ├── bgLogin.jpg
│   │   │   ├── icons
│   │   │   │   ├── bars-solid.svg
│   │   │   │   └── vue-logo.svg
│   │   │   └── logo.svg
│   │   ├── index.html
│   │   ├── index.js
│   │   ├── login.html
│   │   ├── main.css
│   │   ├── training.html
│   │   └── trainings.html
└── index.tsx
```
In my requirements, I need to load this external-app (all static files) inside an `iframe` (or similiars) located in `App.tsx`. This is due to the fact that I need to process some props substitutions inside `external/dashboard-app/index.html` that are dynamically injected from `index.tsx`.

My idea was to leverage webpack loaders to `require.context` the external app and then in some way inject the content directly inside the iframe.

```typescript
const loadAll = (ctx: __WebpackModuleApi.RequireContext) => {
  let keys = ctx.keys();
  let filesContent = keys.map(ctx);
  return keys.reduce((o: Record<string, any>, path, i) => {
    o[path] = filesContent[i];
    return o;
  }, {});
};
var allFiles = loadAll(require.context("./external/dashboard-app/", true, /.*/));
```

Is this feasible or are there better ways/approaches?

**Comment by vankop** [comment]
> Is this feasible or are there better ways/approaches?

depending on your requirements.. yes, you can use `require.context`

**Reply by browser-bug** [reply]
Thanks for the reply. I'm having some troubles pursuing this road. Could you be more detailed on how would you proceed on the task if possible?

**Reply by vankop** [reply]
code snippet in https://github.com/webpack/webpack/discussions/16016#discussion-4190191 looks good
